pkg/template/controller/templateinstance_controller.go

```diff
@@ -51,9 +53,11 @@ const readinessTimeout = time.Hour
 // using its own service account, first verifying that the requester also has
 // permissions to instantiate.
 type TemplateInstanceController struct {
-	restmapper     meta.RESTMapper
-	config         *rest.Config
-	templateClient templateclient.Interface
+	restmapper        meta.RESTMapper
+	dynamicRestMapper *discovery.DeferredDiscoveryRESTMapper
+	config            *rest.Config
```

Comment:
why do you have two configs?

Reply:
one is for json and one is not...  we added the json one to handle unstructured content.  the bulk mapper for checkReadiness is still using the non-json one:

```
		cli, err := bulk.ClientMapperFromConfig(c.config).ClientForMapping(mapping)
```

is it safe to switch it to the json based config?

Reply:
additional context on what we do w/ the client we retrieve:
```
		obj, err := cli.Get().Resource(mapping.Resource).NamespaceIfScoped(object.Ref.Namespace, mapping.Scope.Name() == meta.RESTScopeNameNamespace).Name(object.Ref.Name).Do().Get()
```